src/rpc/methods/eth.rs

```diff
@@ -34,11 +49,54 @@ pub const ETH_BLOCK_NUMBER: &str = "Filecoin.EthBlockNumber";
 pub const ETH_CHAIN_ID: &str = "Filecoin.EthChainId";
 pub const ETH_GAS_PRICE: &str = "Filecoin.EthGasPrice";
 pub const ETH_GET_BALANCE: &str = "Filecoin.EthGetBalance";
+pub const ETH_GET_BLOCK_BY_NUMBER: &str = "Filecoin.EthGetBlockByNumber";
 pub const ETH_SYNCING: &str = "Filecoin.EthSyncing";
 pub const WEB3_CLIENT_VERSION: &str = "Filecoin.Web3ClientVersion";
 
 const MASKED_ID_PREFIX: [u8; 12] = [0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
 
+const BLOOM_SIZE: usize = 2048;
+
+const BLOOM_SIZE_IN_BYTES: usize = BLOOM_SIZE / 8;
+
+const FULL_BLOOM: [u8; BLOOM_SIZE_IN_BYTES] = [0xff; BLOOM_SIZE_IN_BYTES];
+
+const ADDRESS_LENGTH: usize = 20;
+
+const EVM_WORD_LENGTH: usize = 32;
+
+/// Keccak-256 of an RLP of an empty array
+const EMPTY_UNCLES: &str = "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347";
+
+const EIP_1559_TX_TYPE: u64 = 2;
+
+/// The address used in messages to actors that have since been deleted.
+const REVERTED_ETH_ADDRESS: &str = "0xff0000000000000000000000ffffffffffffffff";
```

Comment:
There's a golden opportunity here for docs! I'd love to learn about more about empty uncles :rofl:

Reply:
Uncles are blocks that do not participate in the main chain, but uncle miners are rewarded nevertheless because they burn some energy during Ethash PoW.
In our case, we pretend there are no uncles whatsoever. We have tipsets!